### DISCUSSION TITLE
Improve naming of `write_screen_file`and `write_scrollback_file`

### DISCUSSION THREAD

**Discussion by 00-kat** [ghostty-org/ghostty#3652] [Feature Requests, Ideas]
Incredibly confusing names; people have gotten confused about it before (e.g. #2040, #3496). The default bindings also use `write_scrollback_file` instead of `write_screen_file` which I don't think is intentional. And here's a conversation where @jcollie and I get confused about what the two do, as well: https://discord.com/channels/1005603569187160125/1005603569711452192/1322424870025302037.

I remember there was talk about improving them in the Discord, but I couldn't find any existing issues/discussions about it, so I decided to make one.

**Comment by Integralist** [comment]
👋🏻 

I came here to ask about `write_scrollback_file` and if someone could give me an example use case as I don't understand its purpose 😬 

I _think_ it is for when you execute a command and you want to have the output automatically written to a temporary file.

For example, back when I was using [Warp](https://www.warp.dev/), who just truncate long output (seriously annoying and won't let you configure it to be larger) I typically had to remember to pipe to tee ` whatever | tee /tmp/output` so if (for example) my very large test suite failed I could see all the relevant output.

So I'm presuming `write_scrollback_file` would do this automatically and save me from having to do any manual piping.

Do I have that correct or am I completely mistaken and it's useful for some other purpose? 😅 

Thanks for any advice/guidance.

**Reply by 00-kat** [reply]
Until #189, `write_screen_file` is the easiest way to search for text on the screen.

**Comment by bulletmark** [comment]
I quite like the simple idea of `vim ctrl+shift+j` to easily be able to search and/or mine my screen history. However, I find it very odd that the current screen is not included in the scroll-back record. Then I found issue #3496 and this discussion to discover `write_screen_file` exists. Surely that is what 99% of users would prefer and thus should be the default for ` ctrl+shift+j`?

**Reply by 00-kat** [reply]
That's what I had thought too.  I actually can't determine what the purpose of `write_scrollback_file` is...

**Comment by 00-kat** [comment]
N.B.: the default binding was now changed to use `write_screen_file`. #5285
